### PR TITLE
stats: report correctly CPU usage

### PR DESCRIPTION
**Pull request by giuseppe**
Closes: https://github.com/containers/libpod/issues/4409
    
Signed-off-by: Giuseppe Scrivano <giuseppe@scrivano.org>

### CODE REVIEW

**Comment by rhatdan** [comment]
Can you add tests for this. So we avoid regressions in the future?

File: libpod/stats.go [hunk 1]
```diff
@@ -98,20 +97,15 @@ func getMemLimit(cgroupLimit uint64) uint64 {
 	return cgroupLimit
 }
 
-func calculateCPUPercent(stats *cgroups.Metrics, previousCPU, previousSystem uint64) float64 {
+func calculateCPUPercent(stats *cgroups.Metrics, previousCPU, now, previousSystem uint64) float64 {
```
Comment (mheon) [inline libpod/stats.go]: Can you add an explanation of what the parameters are?

**Comment by giuseppe** [comment]
> Can you add tests for this. So we avoid regressions in the future?

I've tried to add one, but it seems quite tricky.  `stats` need to run at least for two iterations to get some CPU usage data, while `--no-stream` blocks on the first iteration and there is no CPU data available yet.

**Comment by rhatdan** [comment]
LGTM
@mheon @vrothberg @baude @jwhonce @QiWang19 @TomSweeneyRedHat PTAL

**Pull request merged**